src/js/actions/MyProjects/ProjectLoadingActions.js

```diff
@@ -217,7 +217,18 @@ export function displayTools() {
  * prevent a new project from loading
  */
 export function clearLastProject() {
-  return (dispatch) => {
+  return (dispatch, getState) => {
```

Comment:
@neutrinog So the fix was moving the code to disconnect the tools from componentWillUnmount to here?

Reply:
lol nvm just read the description